proposals/multi-miner-worker-pool.md

```diff
@@ -0,0 +1,109 @@
+benefit from this pattern. Small miners could run with a node just large enough to compute window posts, and
+have zero worker nodes until needed. Miners like this could scale up when necessary and scale back down. However,
+I suspect the additional complexity for running a separate broker may negate any benefit for smaller miners.
+
+
+#### Assumptions &amp; hypotheses
+
+* Running a mining operation is expensive
+* Multi-node mining topology is relatively rigid
+* Mining operators want to efficiently allocate resources
+
+#### User workflow example
+
+1. Install message broker of choice. Leave the choice up to the user.
+2. Configure lotus-miner to pass work to amqp://<broker> or maybe mqtt://<broker> NOT to the worker
+3. Configure lotus-worker to subscribe to the broker NOT to the miner
+4. Setup lotus-worker to scale with their cloud provider (optional)
+
+
+#### Impact
+A work-queue would simplify running lotus on large-scale operations. Miners who experience high operational
+cost and toil would appreciate the ease with which brokered work queues scales compared to the monolithic model.
+
+#### Leverage
+I suspect this project is low-leverage because it would target larger miners, but I have no reasonable estimation.
+
+#### Confidence
+
+Impact: 5 -- I'm not sure. The purpose for this proposal, in addition to easing our own future mining aspirations, is to increase
+retention by making it easier to run lotus at at large scale
+Confidence: 0.3 -- There are plenty of worker-queue applications that work well. None of them I know of are Filecoin implementations.
+Ease: 7 -- This chainge would require changes to the miner, worker, and perhaps the protocol they use to communicate. It's not clear to me whether it's better transmit sectors in-band or assume the existance of a shared volume.
+
+Overall: 4.1
+
+## Project definition
+#### Brief plan of attack
+
+Write a testground plan to simulate idle workers under different load conditions. This simulation will be used to measure success.
+* Implement proof of concept.
+* Using the proof of concept, test for scalability and try to extrapolate cost savings estimations for miners of different size and configuration.
+* learn and improve toward a production implementation.
+
+#### What does done look like?
+Brokered pubsub queue between miner and worker
+
+####  What does success look like?
+Worker nodes are 100% utilized even with uneven miner allocation.
+
+#### Counterpoints &amp; pre-mortem
+I'm not confident that this is the biggest problem big miners have.
+It will be more complex to maintain two different kinds of job scheduling.
+
+#### Alternatives
+* Workers could be made to connect to multiple miners simultaneously. Operators with multiple miners would try to schedule on workers with the least-load, or round-robin, hash-ring or some other mechanism and re-try when there is too much load on the worker.
```

Comment:
The storage/sealing logic is written in a way which should support accepting tasks from multiple miners, even on separate networks with different sector sizes

It should be as simple as making it possible to run the "storage manager" piece as a standalone process, and making it possible to tell lotus-miner to use an external storage manager  (or even multiple manager instances, giving us worker grouping for free)